encrypted-media-respec.html

```diff
@@ -606,7 +606,83 @@
+                                <var>candidate configuration</var>.
+                              </p>
+                            </li>
+                            <li>
+                              <p>
+                                Set the <a def-id="option-distinctiveIdentifier"></a> member of
+                                <var>modified candidate configuration</var> to <var>distinctive identifier state</var>.
+                              </p>
+                            </li>
+                            <li>
+                              <p>
+                                Let <var>supported configuration</var> be the result of executing the
+                                <a def-id="get-supported-configuration-and-consent-algorithm"></a> algorithm
+                                with <var>implementation</var>, <var>modified candidate configuration</var> and <var>origin</var>.
+                              </p>
+                            </li>
+                            <li>
+                              <p>
+                                If <var>supported configuration</var> is not <code>ConsentDenied</code>,
+                                return <var>supported configuration</var>.
+                              </p>
+                            </li>
+                          </ol>
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          Return <code>null</code>.
+                        </p>
+                      </li>
+                    </ol>
+                  </dd>
+                  <dt>Otherwise:</dt>
+                  <dd>
+                    <ol>
+                      <li>
+                        <p>
+                          Let <var>supported configuration</var> be the result of executing the
+                          <a def-id="get-supported-configuration-and-consent-algorithm"></a> algorithm
+                          with <var>implementation</var>, <var>modified candidate configuration</var> and <var>origin</var>.
+                        </p>
+                      </li>
+                      <li>
+                        <p>If <var>supported configuration</var> is <code>ConsentDenied</code> return <code>null</code>.</p>
+                      </li>
+                      <li>
+                        <p>Return <var>supported configuration</var>.</p>
+                      </li>
+                    </ol>
+                  </dd>
+                </dl>
+              </li>
+            </ol>
+          </section>
+          <section>
+            <h5>Get Supported Configuration and Consent</h5>
+            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration, <code>null</code>, or <code>ConsentDenied</code> as appropriate.
```

Comment:
Although this allows the call sites to be shorter (there is only one condition to handle because null and valid configurations both get forwarded), I wonder if we should be consistent in how we handle the unsuccessful cases by returning `NotSupported` rather than `null`. That would make _this_ algorithm easier to read. It might also make it more clear what is happening at the call sites and that `null` is a possible value when we "return _supported configuration_."